tests/components/sql/__init__.py

```diff
@@ -46,12 +46,26 @@
 
 ENTRY_CONFIG_INVALID_QUERY = {
```

Comment:
Maybe also good to use an example with `WITH` in the tests?

Reply:
As above make also cte tests with both select and as example update.

Reply:
Added at https://github.com/home-assistant/core/pull/105921/commits/5eb739bac5456310d12a5d9d65b9855a5e193ca1